addons/hr_expense/models/hr_expense.py

```diff
@@ -415,6 +415,9 @@ def _compute_price_unit(self):
             else:
                 expense.price_unit = expense.company_currency_id.round(expense.total_amount / expense.quantity) if expense.quantity else 0.
 
+    def _needs_price_unit_recalculation(self):
+        return False
```

Comment:
```suggestion
    def _needs_price_unit_recalculation(self):
        # Hook to be overridden.
        self.ensure_one()
        return self.product_has_cost
```
You're calling it on a single record, so better make sure it's the case.
A small comment doesn't hurt either :wink: 
Also, you don't need to check `product_id` here since it's already done in `product_has_cost` :blush: